src/routes/exercise/controllers.ts

```diff
@@ -21,19 +22,31 @@ const getAllExercises = async (req: Request, res: Response) => {
 };
 
 const createExercise = async (req: Request, res: Response) => {
-  // TO-DO: Middleware to validate the data and structure the info.
-  const createdExercise = await prisma.exercise.create({
-    data: req.body,
-    include: {
-      links: true,
-    },
-  });
+  try {
+    const { error, value } = exerciseSchema.validate(req.body);
 
-  return res.status(201).json({
-    message: getActionSuccessMsg('Exercise', 'created'),
-    data: createdExercise,
-    error: false,
-  });
+    if (error) {
+      return res.status(500).json({ error: error.details[0].message });
+    }
```

Comment:
I think that the error validation is being returned here. WDYT? Also, remember to maintain the same structure for sending the response:

```
data: any
error: boolean
message: string
```